docs/developers/architecture.md

```diff
@@ -0,0 +1,147 @@
+    parsedres2-->ResourceInputConversionFailure: transformation failure
+}
+pywork-->valdata
+state valdata {
+    parsedres3-->ResourceDeserialised: transformations
+}
+```
+
+### Parsing XML Files
+
+<!-- markdownlint-disable MD013 -->
+
+```mermaid
+---
+title: Parsing of XML File and Transformations Into ParsedResource
+---
+stateDiagram-v2
+state "Transform etree Root" as eroot
+state "Transform etree Root into Python Representation" as transpy
+state "Resource etree" as resetree1
+state "Resource etree" as resetree2
+state "ParsedResource" as respars
+state "Value etree" as valtree
+state "IIIF/bitstream etree" as filetree
+state "ParsedValue" as valpars
+state "ParsedFileValue" as filepars
+state "Transform Resource" as transres
+state "Transform Value" as transval
+state "Transform FileValue" as transfile
+state "Transformed Content" as rescontent
+
+[*]-->r1: Parse file
+
+r4-->transpy
+    state eroot {
+        state "etree root" as r1
+        state "etree root" as r2
+        state "etree root" as r3
+        state "etree root" as r4
+        r1-->r2: Remove Comments
+        r2-->[*]: xsd validation failure
+        r2-->r3: xsd validation success
+        r3-->r4: make localnames
+    }
+    state transpy {
+        resetree1-->transres
+        resetree1-->transval: extract values
+        resetree1-->transfile: extract file or iiif
+        
+        state transres {
+            resetree2-->rescontent: transform special tags</br></br>resolve prefixes into absolute IRIs
+        }
+        transres-->respars
+
+        state transval {
+            valtree-->valpars: transform special tags</br></br>resolve prefixes into absolute IRIs</br></br>map value type into Python Class
+        }
+        transval-->respars: add to resource
+
+        state transfile {
+            filetree-->filepars: map file type
+        }
+        transfile-->respars: add to resource
+    }
+```
+
+<!-- markdownlint-enable MD013 -->
+
+
+### From `ParsedResource` to `IntermediaryResource` in `xmlupload`
+
+```mermaid
+---
+title: Transformations from ParsedResource for xmlupload
+---
+stateDiagram-v2
+
+state "ParsedResource" as start
+state "Transform Entire Resource" as transformall
+state "Transform Resource" as transformres
+state "Transform Value" as transfomrationval
+state "Transform FileValues" as transformfile
+state "ParsedValue" as parsedval
+state "ParsedResource" as parsedres
+state "IntermediaryValue" as valdes
+state "Collected Transformations" as coll
+
+start-->transformall
+
+state transformall {
+    parsedres-->transformfile
+    parsedres-->transfomrationval
+    parsedres-->transformres
+    state transformres {
+        ParsedResource-->Permissions: resolve permissions
+    }
+    state transfomrationval {
+        parsedval-->valdes: map permissions</br></br>map listnodes to IRIs</br></br>map file metadata
+    }
+    state transformfile {
+        ParsedFileValue-->IntermediaryFileValue: map permissions</br></br>map metadata
```

Comment:
Here you say "map permissions", but on the resource level, it's "resolve permissions". Shouldn't this be the same both times?